api/src/main/java/org/pmiops/workbench/jira/JiraService.java

```diff
@@ -0,0 +1,163 @@
+ *
+ * <p><code>
+ * # Get USER / API KEY from credentials bucket.
+ * curl --request GET \
+ *   --url "https://precisionmedicineinitiative.atlassian.net/rest/api/3/issue/createmeta?projectKeys=RW&expand=projects.issuetypes.fields" \
+ *   --header Accept: application/json \
+ *   --user "USER:API_KEY" | jq .
+ *  </code>
+ */
+@Service
+public class JiraService {
+  private static final String PROJECT_KEY = "RW";
+  private static final Logger log = Logger.getLogger(JiraService.class.getName());
+
+  @Autowired private Provider<JiraApi> apiProvider;
+
+  public enum IssueProperty {
+    DESCRIPTION("description"),
+    EGRESS_VM_PREFIX("Egress VM Prefix[Short text]", "customfield_10795"),
+    ISSUE_TYPE("issuetype"),
+    LABELS("labels"),
+    PROJECT("project"),
+    RW_ENVIRONMENT("RW Environment[Short text]", "customfield_10759"),
+    SUMMARY("summary");
+
+    private final String key;
+    private final String fieldName;
+
+    IssueProperty(String key) {
+      this(key, key);
+    }
+
+    IssueProperty(String key, String fieldName) {
+      this.key = key;
+      this.fieldName = fieldName;
+    }
+
+    public String key() {
+      return key;
+    }
+
+    public String fieldName() {
+      return fieldName;
+    }
+  }
+
+  public enum IssueType {
+    TASK("Task", "10001"),
+    BUG("Bug", "10003"),
+    STORY("Story", "10000");
+
+    private final String jiraName;
+    private final String jiraId;
+
+    IssueType(String jiraName, String jiraId) {
+      this.jiraName = jiraName;
+      this.jiraId = jiraId;
+    }
+
+    public String jiraName() {
+      return jiraName;
+    }
+
+    public String jiraId() {
+      return jiraId;
+    }
+  }
+
+  public CreatedIssue createIssue(
+      IssueType type, String description, Map<IssueProperty, Object> issueProps)
+      throws ApiException {
+
+    Map<IssueProperty, Object> mergedProps = new HashMap<>(issueProps);
+    mergedProps.putAll(
+        ImmutableMap.<IssueProperty, Object>builder()
+            .put(IssueProperty.PROJECT, new Project().key(PROJECT_KEY))
+            .put(IssueProperty.ISSUE_TYPE, new IssueTypeDetails().id(type.jiraId()))
+            .put(IssueProperty.DESCRIPTION, textAsMinimalAtlassianDocument(description))
+            .build());
+    try {
+      return apiProvider
+          .get()
+          .createIssue(
+              new IssueUpdateDetails()
+                  .fields(
+                      mergedProps.keySet().stream()
+                          .collect(Collectors.toMap(IssueProperty::fieldName, mergedProps::get))),
+              false);
+    } catch (ApiException e) {
+      logJiraErrorPayload(e);
+      throw e;
+    }
+  }
+
+  public void commentIssue(String issueId, String body) throws ApiException {
+    try {
+      apiProvider
+          .get()
+          .addComment(new Comment().body(textAsMinimalAtlassianDocument(body)), issueId);
+    } catch (ApiException e) {
+      logJiraErrorPayload(e);
+      throw e;
+    }
```

Comment:
nit Avoid log-and-(re)throw:  [go/java-practices/exceptions#log_rethrow](go/java-practices/exceptions#log_rethrow)

Reply:
Yeah, this is not ideal. I added a comment below to explain why I'm doing this. Let me know if you have other ideas.

I think it's correct to throw a checked exception from these messages, but I also don't want to have anyone who catches this need special logic to get at the underlying error message details.

Reply:
SG! log & rethrow seems to be the easiest solution..